recipes-extended/xen/files/libxl-xenmgr-support.patch

```diff
@@ -170,22 +170,26 @@ PATCHES
  out:
 --- a/tools/libxl/libxl_dm.c
 +++ b/tools/libxl/libxl_dm.c
-@@ -2118,6 +2118,20 @@ retry_transaction:
+@@ -2118,6 +2118,24 @@ retry_transaction:
      libxl__xs_printf(gc, XBT_NULL,
                       DEVICE_MODEL_XS_PATH(gc, dm_domid, guest_domid, "/xen_extended_power_mgmt"), "2");
  
 +    /* OpenXT: We wait on xenmgr writing v4v-firewall-ready.
 +     *         xenmgr waits on us writing image/device-model-domid
 +     */
-+    int32_t timeout = 0;
-+    char * ready = NULL;
++    unsigned int timeout;
++    char *ready = NULL;
++    const char *firewall_path =
++        GCSPRINTF("%s/v4v-firewall-ready",
++                  libxl__xs_get_dompath(gc, guest_domid));
 +    /* Block and wait for v4v firewall rules */
-+    while (timeout < 30) {
-+        ready = libxl__xs_read(gc, XBT_NULL, GCSPRINTF("%s/v4v-firewall-ready", libxl__xs_get_dompath(gc, guest_domid)));
-+        if(ready)
++    for (timeout = 0; !ready && timeout < 4; ++timeout) {
```

Comment:
I agree 30 seconds is excessive, and I agree the stubdom should boot in 4 seconds or less.  But do we want a little more margin just in case?  Or do we normally boot in 2 seconds (guessing) so 4 seconds is plenty?

Reply:
I picked 4s because it is usually, as you guessed, up around 2s on the machine I tested. The stubdom timeout is defined to 30s and based on an `xs_watch`, we could possibly index on that?

This is a little on the exotic side of things I guess. Maybe it could also be refactored in an AO along the stubdom creation path, but that should be tackled in a different PR.